library/core/src/intrinsics/simd.rs

```diff
@@ -232,6 +245,9 @@ extern "platform-intrinsic" {
     /// corresponding value in `val` to the pointer.
     /// Otherwise if the corresponding value in `mask` is `0`, do nothing.
     ///
+    /// The stores happen in left-to-right order.
+    /// (This is relevant in case two of the stores overlap.)
```

Comment:
( probably not relevant, but fun facts to know and tell: if the machine instruction gets executed, this guarantee also affects the state of the CPU during exception handling. )

Reply:
Wait, exception handles can observe these instructions as non-atomic? Wow.

Reply:
Yeah, on Intel CPUs at least, for most variants of these instructions, the operands (esp. the mask operand) get updated so that continuing with the same operands will complete the operation, completing a single store for each index.

Reply:
What do you mean "get updated"? Does this change the contents of some other register, where the mask operand is stored? I hope ti restored the original value when the instruction is done?

Reply:
> What do you mean "get updated"? Does this change the contents of some other register, where the mask operand is stored?

Yes, the mask register.

> I hope it restored the original value when the instruction is done?

uhhh probably! :grin:

Reply:
> > I hope it restored the original value when the instruction is done?
> 
> uhhh probably! 😁

nope, it's defined to clear the mask register: https://www.felixcloutier.com/x86/vscatterdps:vscatterdpd:vscatterqps:vscatterqpd

Reply:
Okay... well as long as LLVM takes that into account it's all good.